### PR TITLE
hipify caffe2 script in fbcode

### PR DESCRIPTION
**Pull request by xw285cornell**
Summary:
Make changes to make hipify_python script to work with fbcode.

1. Add TARGETS file
2. Make hipify_python a module as well as a standalone script.

Differential Revision: D10851216

### CODE REVIEW

File: tools/amd_build/build_pytorch_amd.py [hunk 1]
```diff
@@ -1,9 +1,7 @@
-import shutil
 import subprocess
 import os
-import sys
-from shutil import copytree, ignore_patterns
 from functools import reduce
+from pyHIPIFY import hipify_python
```
Comment (bddppq) [inline tools/amd_build/build_pytorch_amd.py]: This probably doesn't work with python3 in OSS, maybe better to add sys.path

**Review by bddppq — APPROVED** [review]

File: tools/amd_build/pyHIPIFY/hipify_python.py [hunk 2]
```diff
@@ -45,6 +45,33 @@
 CAFFE2_TEMPLATE_MAP = {}
 
 
+class Error(Exception):
```
Comment (bddppq) [inline tools/amd_build/pyHIPIFY/hipify_python.py]: maybe simply 
```
class Error(Exception): pass
class InputErorr(Error): pass
```
?

File: tools/amd_build/pyHIPIFY/hipify_python.py [hunk 3]
```diff
@@ -45,6 +45,33 @@
...
+    """Base class for exceptions in this module."""
+
+    def __init__(self, expression, message, desc):
```
Comment (bddppq) [inline tools/amd_build/pyHIPIFY/hipify_python.py]: The only place uses InputError is only passing one argument:
https://github.com/pytorch/pytorch/blob/d05b4e31d1a41bf5c908adb2d3a0d4d6ba0118fc/tools/amd_build/pyHIPIFY/hipify_python.py#L422

File: tools/amd_build/pyHIPIFY/hipify_python.py [hunk 4]
```diff
@@ -695,7 +727,7 @@ def disable_function(input_string, function, replace_style):
     return output_string
 
 
-def get_hip_file_path(filepath, hipify_caffe2):
+def get_hip_file_path(filepath, hipify_caffe2, hip_suffix):
```
Comment (bddppq) [inline tools/amd_build/pyHIPIFY/hipify_python.py]: I suggest let's use ".hcc" everywhere

Comment (xw285cornell) [reply tools/amd_build/pyHIPIFY/hipify_python.py]: Yes I'll change to hcc everywhere :)

**Pull request closed without merging**